GeneralsMD/Code/GameEngine/Include/GameLogic/AI.h

```diff
@@ -312,7 +312,9 @@ class Team;
 class Weapon;
 
 // Note - written out in save/load xfer and .map files, don't change these numbers.  
-enum AttitudeType { AI_SLEEP = -2, AI_PASSIVE=-1, AI_NORMAL=0, AI_ALERT=1, AI_AGGRESSIVE=2, AI_INVALID=3 };		///< AI "attitude" behavior modifiers
+
+// TheSuperHackers @compile Renames AI_PASSIVE to not conflict with macro in ws2def.h
+enum AttitudeType { AI_SLEEP = -2, AI_Passive=-1, AI_NORMAL=0, AI_ALERT=1, AI_AGGRESSIVE=2, AI_INVALID=3 };		///< AI "attitude" behavior modifiers
```

Comment:
Not a fan of introducing an inconsistency like this. Recommend renaming all of these, something like SAGE_AI_xxx.

Reply:
We can make them all PascalCase maybe?
So have `AI_Sleep`, `AI_Passive`, ...?

I am not too fond of the `SAGE_` prefix unless there is precedence for it anywhere else in this code.

Reply:
We could, yes. My proposal was to sidestep casing entirely, but if you prefer pascal case for everything, we can define that now.

Reply:
How about ATTITUDE_SLEEP, ATTITUDE_PASSIVE, ... No idea why it has AI prefix in the first when the enum name is AttituteType. The other enum names in this file are also upper case.

Reply:
Renamed to `ATTITUDE_*`.

Reply:
It might be a more minimalist change to just `#undef AI_PASSIVE` if it's defined, until platform headers (i.e: `ws2def.h`) are properly isolated.

Reply:
I did experiment with that first but found no satisfying way to do that. Winsock spills in there some way. We would need to isolate the winsock include to get rid of it. It is unclear to me how complex that is.

If someone can show a sample implementation of getting rid of it cleanly (without `#undef`, because that just masks problems), then I can revisit this.